resources/components/InputFilter.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+  <div class="flex flex-row py-5 mx-auto justify-center">
+    <input
+      class="ml-10 py-3 pl-2 w-full bg-gray-800 border-blue-300 rounded-l-lg text-blue-300"
+      placeholder="Search"
+      v-model="searchInput"
+      @input="processAndSendData(dataToFilter, searchInput)"
+    />
+  </div>
+</template>
+
+<script>
+import { properties } from '../mocks/api';
+
+export default {
+  name: 'table-filter',
+  props: {
+    dataToFilter: {
+      type: Array,
+      default: () => [],
+    },
+  },
+  data() {
+    return {
+      properties,
+      searchInput: '',
+      processedData: [],
+    };
+  },
+  watch: {
+    searchInput(val) {
+      if (!val.length) {
+        this.$emit('filter-by-input', properties, false);
+        this.$emit('reloadTable', true);
+      }
+    },
+  },
+  methods: {
+    processAndSendData(arr, input) {
```

Comment:
¿Por qué enviaste las propiedades vía props al método y no la usaste directamente teniéndolas accesibles instancia la instancia?: "this.dataToFilter" y "this.searchInput"